src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -506,6 +491,63 @@ private static StringBuilder fakeBody(final Request... dispositions)
                 .append(String.format("--%s--", BOUNDARY));
             return builder;
         }
+
+        /**
+         * This class is  for the specific implementation
+         * for the RqMultipart, (see the overridden methods for details).
+         *
+         * @todo #517:30min re-design the class.
+         *  Why the super class is encapsulated inside the class?
+         *  Find an implementation which won't be using
+         *  both composition and inheritance of the same class.
+         *  Either get the header values from the super class,
+         *  or if the purpose is to have a one-to-many header to body relation,
+         *  create the class without extending Request.
+         */
+        private static class RqMultipartRequest implements Request {
+
+            /**
+             * Request object. Holds value for the header.
```

Comment:
@sebing now I'm thinking... It's just normal request. It doesn't have any specific value for the header.
My proposition is to change Javadoc to `Request to be decorated with multipart header.`